compiler/testData/diagnostics/tests/EnumEntryAsType.fir.kt

```diff
@@ -10,7 +10,7 @@ enum class Color {
 class MyColor(val x: <!UNRESOLVED_REFERENCE!>Color.RED<!>, y: <!UNRESOLVED_REFERENCE!>Color.RED<!>) : <!UNRESOLVED_REFERENCE!>Color.RED<!> {
 
     var z: <!UNRESOLVED_REFERENCE!>Color.RED<!> = Color.RED
-    set(arg: <!UNRESOLVED_REFERENCE!>Color.RED<!>) { z = arg }
+    set(arg: <!UNRESOLVED_REFERENCE, WRONG_SETTER_PARAMETER_TYPE!>Color.RED<!>) { z = arg }
```

Comment:
This shouldn't be type mismatch: https://github.com/JetBrains/kotlin/blob/master/compiler/testData/diagnostics/tests/EnumEntryAsType.kt#L13